wgpu-core/src/device/mod.rs

```diff
@@ -420,6 +433,17 @@ pub enum DeviceError {
     DeviceMismatch(#[from] Box<DeviceMismatch>),
 }
 
+impl AsWebGpuErrorType for DeviceError {
+    fn as_webgpu_error_type(&self) -> ErrorType {
+        match self {
+            DeviceError::DeviceMismatch(e) => e.as_webgpu_error_type(),
+            Self::Invalid(_) => ErrorType::Validation,
+            Self::Lost | Self::ResourceCreationFailed => ErrorType::Internal,
```

Comment:
In servo we have this: https://github.com/servo/servo/blob/ba061ec2b0ef7124a5e64ec11a406cbc45cac02f/components/webgpu/wgpu_thread.rs#L389, I think instead of adding new variant, as_webgpu_error_type could return Option when wgpu error is not considered as error.

Reply:
@sagudev: Internal errors are in the same error hierarchy as OOM and validation errors. Device loss, however, is usually reported through. I need to investigate whether reporting device loss as something distinct from an internal error is still necessary, if that's properly hooked up.

CC @teoxoy and @jimblandy for Firefox thinking.

Reply:
We still need an error type for it I guess but that can also be ignored in downstream code.